src/test/java/net/smartcosmos/dao/metadata/impl/MetadataPersistenceServiceTest.java

```diff
@@ -198,18 +199,16 @@ public void testUpsert() throws Exception {
         assertTrue(Boolean.parseBoolean(response.get().getMetadata().get("upsertBool").toString()));
         assertEquals("Text", response.get().getMetadata().get("upsertString").toString());
 
-        JSONObject output = (JSONObject) response.get().getMetadata().get("upsertJson");
```

Comment:
Why are the `JSONObject` and `JSONArray` tests deleted or changed to use different types? Shouldn't the value parser be able to handle such types?